src/main/java/com/adobe/abp/regola/rules/RangeRule.java

```diff
@@ -0,0 +1,140 @@
+/*
+ *  Copyright 2023 Adobe. All rights reserved.
+ *  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License. You may obtain a copy
+ *  of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software distributed under
+ *  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *  OF ANY KIND, either express or implied. See the License for the specific language
+ *  governing permissions and limitations under the License
+ */
+
+package com.adobe.abp.regola.rules;
+
+import com.adobe.abp.regola.facts.FactsResolver;
+import com.adobe.abp.regola.results.Result;
+import com.adobe.abp.regola.results.RuleResult;
+import com.adobe.abp.regola.results.ValuesRuleResult;
+import com.adobe.abp.regola.utils.futures.FutureUtils;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+/**
+ * A rule that checks if a fact exists between two numbers with various inclusivity options.
+ */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class RangeRule extends KeyBasedRule {
+
+    private final Number min;
+    private final Number max;
```

Comment:
I think it would still be great to have this rule accept other types. You can have a range of any element types as long as the types are partially orderable.
See: https://en.wikipedia.org/wiki/Partially_ordered_set